pkg/admission/validator/shoot.go

```diff
@@ -113,6 +114,14 @@ func (s *shoot) validateShoot(shoot *core.Shoot, oldInfraConfig, infraConfig *ap
 
 	// Network validation
 	allErrs = append(allErrs, azurevalidation.ValidateNetworking(shoot.Spec.Networking, nwPath)...)
+	if shoot.Spec.Networking.Type == calico.ReleaseName {
+		networkConfig, err := decodeCalicoNetworkingConfig(s.decoder, shoot.Spec.Networking.ProviderConfig)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(nwPath.Child("providerConfig"), err))
+		} else if networkConfig.Overlay != nil && networkConfig.Overlay.Enabled {
+			allErrs = append(allErrs, field.Forbidden(nwPath.Child("providerConfig").Child("overlay").Child("enabled"), "Calico overlay network is not supported on azure"))
+		}
+	}
```

Comment:
Can't it be part of the `validateNetworking` function (ln: 116) ?

Reply:
Where should `decodeCalicoNetworkingConfig` go then? Should it be next to `validateNetworking`, where we do not have such functions as of now?

Reply:
You can probably use https://github.com/gardener/gardener-extension-provider-azure/blob/master/pkg/webhook/network/mutate.go#L36 but if not then we can move the function to this file or to a helper.go in the same package.

Reply:
Moved code as proposed. Please take another look.